src/Numerics/ODESolvers/SplitExplicitMethod.jl

```diff
@@ -0,0 +1,192 @@
+            coupled,
+            RT(add_fast_substeps),
+        )
+    end
+end
+
+function dostep!(
+    Qvec,
+    split::SplitExplicitSolver{SS},
+    param,
+    time::Real,
+) where {SS <: LSRK2N}
+    slow = split.slow_solver
+    fast = split.fast_solver
+
+    Qslow = Qvec.slow
+    Qfast = Qvec.fast
+
+    dQslow = slow.dQ
+    dQ2fast = similar(dQslow)
+
+    slow_bl = slow.rhs!.balance_law
+    fast_bl = fast.rhs!.balance_law
+
+    groupsize = 256
+
+    slow_dt = getdt(slow)
+    fast_dt_in = getdt(fast)
+
+    for slow_s in 1:length(slow.RKA)
+        # Current slow state time
+        slow_stage_time = time + slow.RKC[slow_s] * slow_dt
+
+        # Initialize fast model and tentency adjustment before evalution of slow mode
+        if split.coupled
+            initialize_states!(
+                slow_bl,
+                fast_bl,
+                slow.rhs!,
+                fast.rhs!,
+                Qslow,
+                Qfast,
+            )
+        end
+
+        # Evaluate the slow mode
+        # --> save tendency for the fast
+        slow.rhs!(dQ2fast, Qslow, param, slow_stage_time, increment = false)
+
+        # vertically integrate slow tendency to advance fast equation
+        # and use vertical mean for slow model (negative source)
+        # ---> work with dQ2fast as input
+        if split.coupled
+            tendency_from_slow_to_fast!(
+                slow_bl,
+                fast_bl,
+                slow.rhs!,
+                fast.rhs!,
+                Qslow,
+                Qfast,
+                dQ2fast,
+            )
+        end
+
+        # Compute (and RK update) slow tendency
+        slow.rhs!(dQslow, Qslow, param, slow_stage_time, increment = true)
+
+        # Update (RK-stage) slow state
+        event = Event(array_device(Qslow))
+        event = update!(array_device(Qslow), groupsize)(
+            realview(dQslow),
+            realview(Qslow),
+            slow.RKA[slow_s % length(slow.RKA) + 1],
+            slow.RKB[slow_s],
+            slow_dt,
+            nothing,
+            nothing,
+            nothing;
+            ndrange = length(realview(Qslow)),
+            dependencies = (event,),
+        )
+        wait(array_device(Qslow), event)
+
+        # Fractional time for slow stage
+        if slow_s == length(slow.RKA)
+            γ = 1 - slow.RKC[slow_s]
+        else
+            γ = slow.RKC[slow_s + 1] - slow.RKC[slow_s]
+        end
+
+        # RKB for the slow with fractional time factor remove (since full
+        # integration of fast will result in scaling by γ)
+        nsubsteps = fast_dt_in > 0 ? ceil(Int, γ * slow_dt / fast_dt_in) : 1
+        fast_dt = γ * slow_dt / nsubsteps
+
+        updatedt!(fast, fast_dt)
+
+        for substep in 1:nsubsteps
+            fast_time = slow_stage_time + (substep - 1) * fast_dt
+            dostep!(Qfast, fast, param, fast_time)
+            #  ---> need to cumulate U at this time (and not at each RKB sub-time-step)
```

Comment:
wrap

Reply:
ask @jm-c, he wrote it

Reply:
I think it is a very old comment, I've deleted it for now